Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -118,20 +151,21 @@ static inline void setStateScrollingNodeSnapOffsetsAsFloat(ScrollingStateScrolli
     node.setSnapOffsetsInfo(offsetInfo->convertUnits<FloatScrollSnapOffsetsInfo>(deviceScaleFactor));
 }
 
-void AsyncScrollingCoordinator::willCommitTree()
+void AsyncScrollingCoordinator::willCommitTree(FrameIdentifier rootFrameID)
 {
-    updateEventTrackingRegions();
+    updateEventTrackingRegions(rootFrameID);
 }
 
-void AsyncScrollingCoordinator::updateEventTrackingRegions()
+void AsyncScrollingCoordinator::updateEventTrackingRegions(FrameIdentifier rootFrameID)
 {
     if (!m_eventTrackingRegionsDirty)
         return;
 
-    if (!m_scrollingStateTree->rootStateNode())
+    auto it = m_scrollingStateTrees.find(rootFrameID);
```

Comment:
Maybe add `existingScrollingStateTreeForRootFrameID()` and use it here.

Reply:
Fixed.